src/apiHelper.ts

```diff
@@ -14,37 +18,52 @@ export async function getWeatherData(city : CityInterface): Promise<[CurrentWeat
 
 	const url = `${baseUrl}?${params.toString()}`;
 
-	try {
-		const response = await fetch(url);
+	const timeInterval$ = interval(600000); 
 
-		if (!response.ok) {
-			throw new Error('Unable to get response');
-		}
-		const data : WeatherDataResponseInterface = await response.json();
-		if (data?.current?.temperature_2m && data?.current?.temperature_2m && data?.daily?.temperature_2m_max.length>0 && data?.daily?.temperature_2m_min.length>0) {
-			const currentWeatherData: CurrentWeatherDataInterface = {
-				temperature : data.current.temperature_2m,
-				temperatureMax: data.daily.temperature_2m_max[0],
-				temperatureMin: data.daily.temperature_2m_min[0],
-				weatherCode: data.current.weather_code,
-			};
+	return timeInterval$.pipe(
+		startWith(0),
+		switchMap(() => 
+			fromFetch(url).pipe(
+			switchMap(response => fromPromise<WeatherDataResponseInterface>(response.json())),
+			switchMap(data => {
+				if (
+					data?.current?.temperature_2m &&
+					data?.current?.temperature_2m &&
+					data?.daily?.temperature_2m_max.length > 0 &&
+					data?.daily?.temperature_2m_min.length > 0
+				) {
+					const currentWeatherData: CurrentWeatherDataInterface = {
+					temperature: data.current.temperature_2m,
+					temperatureMax: data.daily.temperature_2m_max[0],
```

Comment:
Don't forget to also check temperature_2m_max. and temperature_2m_min

Reply:
You're right, I'll fix it. Thanks

Reply:
I haven't even thought about the API updating in its own time... I'm using a different API so the time will be different, but I'll look into it. Thank you.

Reply:
I've updated it if you could please have a look at it?